### ISSUE TITLE
[Feature]: Allow user to create a new sheet inside a Google Sheet

### ISSUE THREAD

**Issue by tomjose92** [appsmithorg/appsmith#32758]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Summary

I have an App that is tracking lot of daily metrics and updating it to Google Sheets.
Since there is a vast amount of information stored there I would like to segregate it based on months in different sheets.

But I am not able to create a new sheet with the new month’s name dynamically via the Google sheet integration, and hence there is a manual effort needed for me to open the Sheets and add it myself.

### Why should this be worked on?

Currently the Google sheets integration allows users to Create new Spreadsheet dynamically, and insert data inside a sheet. It also allows us to delete the Spreadsheet, Sheet and data inside along with updating data.
The only thing missing from the CRUD operations is Add Sheet to the spreadsheet.
This will give additional flexibility in using this integration, and can support lot of usecases that require a level of dynamicity that using pre-built sheets.

**Comment by ame-appsmith** [comment]
[Another user](https://app.frontapp.com/open/cnv_1auqdgp8?key=qVUkterCZWwTdSKBmwULKyOh-4NnSR9E) requested this feature.

**Comment by saicharan-zemoso** [comment]
Hello @Nikhil-Nandagopal @rohan-arthur , I was checking if I could work on this but I was getting this error while adding google sheets as data source.
![Screenshot from 2024-08-07 09-31-22](https://github.com/user-attachments/assets/0292bea1-19fa-4e61-b4d2-6252ca305897)

**Comment by saicharan-zemoso** [comment]
Hello @rohan-arthur , Yes I did and I am getting the same error in local setup and official release application.

**Comment by rohan-arthur** [comment]
@saicharan-zemoso 
Oh, this is a big problem. Do you mind joining me on a call to debug this? Would be great if we can have a call today, and we may need only a few minutes: https://calendly.com/rohan-arthur/60min

**Comment by saicharan-zemoso** [comment]
yes. But i am unable to book a slot today it is showing free slots on 21,22 aug.

**Comment by sneha122** [comment]
Hi @saicharan-zemoso can you please book a slot today using this [link](https://calendly.com/d/cmdc-y8b-tvy/30min)

**Comment by saicharan-zemoso** [comment]
booked a slot at 5:30, thank you.

**Comment by sneha122** [comment]
@saicharan-zemoso Thanks a lot for reporting this issue immediately, I am discussing this internally with the team, meanwhile you can use `Selected Google Sheets` option and go ahead with authorisation for that.

**Comment by rohan-arthur** [comment]
thank you @saicharan-zemoso for finding the problem with creating gsheets datasource! It is a critical bug, and we are tracking it in #35523 
You can come back to this after that is fixed.

**Comment by saicharan-zemoso** [comment]
No problem! I'm glad I could help.
